services/api/lib/thumbnails.js

```diff
@@ -72,45 +83,78 @@ exports.register = function(server, options, done) {
   // Check if the given page has the lowest id in its parent project
   // If it is, then request a new thumbnail
   function checkPageId(page, tail) {
-    if ( !process.env.THUMBNAIL_SERVICE_URL ) {
-      return tail();
-    }
-
-    server.methods.pages.min([
-      page.project_id
-    ], function(err, result) {
-      if ( err ) {
-        server.log('error', {
-          details: 'Error querying DB for lowest page ID in project: ' + page.project_id,
-          error: err
-        });
-        return tail(err);
+    process.nextTick(function() {
+      if ( !process.env.THUMBNAIL_SERVICE_URL ) {
+        return tail();
       }
-      var row = result.rows[0];
+      server.methods.pages.min([
+        page.project_id
```

Comment:
That's what we determined to be "the thumbnail page" for all projects. in most cases, it's the first page made, unless they delete it.